### PR TITLE
Update releases table with v1.6.3, v1.5.2, v1.4.4

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Updates releases table with latest patch versions on all active
branches.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

xref #2899 #2900 #2901 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

n/a -- docs

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,16 +23,16 @@ documentation].
 
 | Release |  Current Patch  | Release Date |      EOL      |
```
Comment (muvaf) [inline README.md:24]:
```suggestion
| Release |  Current Patch  | Minor Release Date |      EOL      |
```
```suggestion
| Release |  Release Date | Current Patch  |      EOL      |
```

I got confused the last time.

**Review by muvaf — APPROVED** [review]
Thanks!

File: README.md (hunk 1 above)
Comment (hasheddan) [reply README.md:24]: @muvaf are these two options for changing or are you thinking it should read `Releases | Minor Release Date | Current Patch | EOL`? I think I would be most amenable to just the second one: `Releases | Release Date | Current Patch | EOL`, which would just be swapping the two middle columns -- what do you think?

Comment (muvaf) [reply README.md:24]: I provided them as different alternatives. The second one reads better to me, too.

Comment (hasheddan) [reply README.md:24]: Awesome! Thanks :)

**Pull request merged**